models/compositions/dynamic_generator.rb

```diff
@@ -0,0 +1,172 @@
+# frozen_string_literal: true
+
+require "concurrent"
+
+module CommonModels
+    module Compositions
+        # Generic implementation of components that generate dynamic vales on their ports
+        #
+        # A specific dynamic generator task can be generated like a ConstantGenerator.
+        # That is, using the {DynamicGenerator.for_type},
+        # {DynamicGenerator.for_data_service}, or the more generic {DunamicGenerator.for},
+        # which resolves into one the previously mentioned methods.
+        #
+        # The values that are generated can be determined initially via the values
+        # argument. It must be set as a hash with the ports name as keys. Afterwards, the
+        # values can be overwritten by assigning to the values attribute.
+        #
+        # @example create and deploy a task that writes a double value of 0.2 initially,
+        # and can be changed dynamically at any time
+        #   DoubleGenerator = DynamicGenerator.for("/double")
+        #   Syskit.conf.use_ruby_tasks DoubleGenerator => "double_dynamic_gen"
+        #   mission = DoubleGenerator.with_arguments(:values => Hash["out" => 10])
+        #   add_mission(mission)
+        #   mission.values = { "out" => 42 }
+        class DynamicGenerator < Syskit::RubyTaskContext
+            # The write period in seconds
+            argument :period, default: 0.1
+
+            # @api private
+            #
+            # The writing thread
+            attr_reader :write_thread
+
+            attr_reader :dynamic_values
```

Comment:
Since we're revisiting this implementation ... let's break with my bad old habit of defining attr_reader for all instance variables.